### PR TITLE
Switch from request to got

### PR DESCRIPTION
**Pull request by solidgoldpig**

### CODE REVIEW

**Review by umaar — APPROVED** [review]

File: lib/fb-user-filestore-client.js [hunk 1]
```diff
@@ -56,65 +56,94 @@ class FBUserfilestoreClient extends FBJWTClient {
...
+   * Bunyan logger instance
+   *
    * @return {promise<undefined>}
    *
    **/
-  store (userId, userToken, file, policy) {
-    const urlPattern = endpoints.store
+  async store (args, logger) {
+    const {userId, userToken} = args
+    let {file, policy} = args
```
Comment (umaar) [inline lib/fb-user-filestore-client.js:145]: Not sure the benefit of `policy = Object.assign({}, policy)` but if we can remove that, policy could be a const. Maybe it's for cloning purposes?

File: lib/fb-user-filestore-client.js [hunk 2]
```diff
@@ -133,48 +162,67 @@ class FBUserfilestoreClient extends FBJWTClient {
...
-      .then(file => this.store(userId, userToken, file, policy))
+  async storeFromPath (filePath, args, logger) {
+    const file = await readFile(filePath)
+    args.file = file
+    return this.store(args, logger)
+  }
+}
+
+FBUserFileStoreClient.offline = () => {
+  class FBUserFileStoreClientOffline extends FBUserFileStoreClient {
```
Comment (umaar) [inline lib/fb-user-filestore-client.js:210]: Interesting

**Review by umaar — APPROVED** [review]

File: lib/fb-user-filestore-client.js (hunk 2 above)
Comment (umaar) [reply lib/fb-user-filestore-client.js:210]: How do you see this tying into to a user journey?

File: lib/fb-user-filestore-client.js (hunk 1 above)
Comment (solidgoldpig) [reply lib/fb-user-filestore-client.js:145]: Just ensuring that we don't change the policy that was passed in.

File: lib/fb-user-filestore-client.js (hunk 2 above)
Comment (solidgoldpig) [reply lib/fb-user-filestore-client.js:210]: Better than doing it in a happenstance way in the runner, having to create stubs for all the methods that will be invoked. I think, anyway...

**Comment by solidgoldpig** [comment]
Actually, happy to bikeshed the name of the `offline` method.

I considered withoutBackend/Connection / mocked (though that's definitely wrong)

**Pull request merged**